### PR TITLE
Kieker 1745 cleanup and deprecation

### PR DESCRIPTION
**Pull request by rju**
# Pull Request

## Contribution
- Finally fixed issue with command line parsing for TraceAnalysisTool with JCommander
- Added additional package-info.java files, which will become more relevant for Java 9

## Code Quality Thresholds
- Could be lowered, fixed some checkstyle issues.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -511,6 +511,8 @@ task apidoc(type: Javadoc) {
     classpath = files(mainSubprojects.collect { project -> project.sourceSets.main.compileClasspath })
     destinationDir = new File(projectDir, 'docs')
 
+    failOnError = false
```
[the remark this answers is not in the document]
Comment (rju) [reply build.gradle]: This is a mistake. I'll remove it. However, I usually get errors with apidocs locally.

File: kieker-common/gradle.properties [hunk 2]
```diff
@@ -3,7 +3,7 @@ checkstyleWarningThreshold = 3
 
 
 findbugsErrorThreshold = 0
-findbugsWarningThreshold = 4
+findbugsWarningThreshold = 48
```
[the remark this answers is not in the document]
Comment (rju) [reply kieker-common/gradle.properties]: A shite. Looks like a record regeneration issue (used wrong target API setting).

[the remark this answers is not in the document]
Comment (rju) [reply kieker-common/gradle.properties]: Fixed this issue

**Comment by rju** [comment]
> Since the last review, many of the previously deleted _hashCode_ methods reappeared. Is this intended?

Yes this is intended. Actually, their removal was accidentally.

**Comment by duelle** [comment]
> > Since the last review, many of the previously deleted _hashCode_ methods reappeared. Is this intended?
> 
> Yes this is intended. Actually, their removal was accidentally.

Thanks for the clarification. 
As soon as the Jenkins builds are successful, I will merge the PR.

**Pull request merged**